# Chapter 3

In [1]:
import numpy as np
import matplotlib.pyplot as plt

<a id="contents"></a>

## Contents

<a href=#ex3.1>Exercise 3.1</a><br>
<a href=#ex3.2>Exercise 3.2</a><br>
<a href=#ex3.3>Exercise 3.3</a><br>
<a href=#ex3.4>Exercise 3.4</a><br>

<a id="ex3.1"></a>

### Exercise 3.1

$P(die A | seq) = \frac{P(seq | die A)P(die A)}{P(seq)}$ <br><br>
$P(die A) = P(die B) = 0.5$ <br><br>
$P(seq | die A) = P(5 | die A) P(3 | die A) P(9 | die A) P(3 | die A) P(8 | die A) P(4 | die A) P(7 | die A)$ <br><br>
$P(seq) = P(die A) P(seq | die A) + P(die B) P(seq | die B)$

In [15]:
p_A = 0.5
p_B = 0.5
p_seq_A = np.product(np.array([1, 3, 1, 3, 1, 2, 1])/20)
p_seq_B = np.product(np.array([2, 2, 1, 2, 2, 2, 2])/20)

In [16]:
(p_A * p_seq_A) / (p_A * p_seq_A + p_B * p_seq_B)

0.21951219512195116

<a href=#contents>Back to contents</a>

<a id="ex3.2"></a>

### Exercise 3.2

Same but with another die which is uniform.

In [19]:
dice_probs = np.array([1/3] * 3)
p_seq_C = np.product(np.array([1/20] * 7))

In [20]:
p_seqs = np.array([p_seq_A, p_seq_B, p_seq_C])
for die, p_seq in zip(["A", "B", "C"], p_seqs):
    p_die = p_seq / np.sum(p_seqs)
    print("Probability of die ", die, ": ", p_die)

Probability of die  A :  0.21686746988
Probability of die  B :  0.771084337349
Probability of die  C :  0.0120481927711


<a href=#contents>Back to contents</a>

<a id=ex3.3></a>

### Exercise 3.3

$P(a<x<b | \lambda) = \int_{a}^{b} \lambda e^{-\lambda x} dx$ <br><br>
$P(\lambda) = 1$ <br><br>


<a href=#contents>Back to contents</a>

<a id="ex3.4"></a>

### Exercise 3.4

$P(Oliver | O, AB) = \frac{P(O, AB | Oliver) P(Oliver)}{P(O, AB)}$ <br><br>
$P(O, AB | Oliver) = P(O | Oliver) P(AB | Oliver) = 1·0.01$ <br><br>
$P(O, AB) = P(O) P(AB) + P(AB) P(O) = 0.6·0.01 + 0.01·0.6$

In [28]:
0.01/(0.6*0.01*2)

0.8333333333333334

$P(Oliver | O, AB) = 0.833 P(Oliver) \rightarrow$ The probability of Oliver being in the crime scene decreases with respect to the prior.

<a href=#contents>Back to contents</a>